sourcefiles/ctoptions.py

```diff
@@ -0,0 +1,604 @@
+'''
+Module for preconfiguring in-game options at compile time
+'''
+
+import byteops
+from ctenums import ActionMap, InputMap, LocID
+from ctrom import CTRom
+import ctevent
+
+class ControllerBinds:
+
+    BUTTONS_OFFSET = 0x02FCA9
+    BUTTONS_LENGTH = 8
+
+    def __init__(self, data: bytearray = None):
+
+        self._mappings = self.get_vanilla()
+        
+        if data is None:
+            return
+
+        self.update_from_bytes(data[-8:])
+
+    @classmethod
+    def get_vanilla(cls):
+
+        ret = {
+            ActionMap.CONFIRM: InputMap.A_BUTTON,
+            ActionMap.CANCEL: InputMap.B_BUTTON,
+            ActionMap.MENU: InputMap.X_BUTTON,
+            ActionMap.DASH: InputMap.B_BUTTON,
+            ActionMap.MAP: InputMap.SELECT_BUTTON,
+            ActionMap.WARP: InputMap.Y_BUTTON,
+            ActionMap.PG_UP: InputMap.R_SHOULDER,
+            ActionMap.PG_DN: InputMap.L_SHOULDER
+        }
+        
+        return ret
+
+    def from_rom(self, ctrom: CTRom):
```

Comment:
Trying to use this method gave errors. BUTTONS_OFFSET and BUTTONS_LENGTH need to be accessed through self.  But really this should be a classmethod instead.  Will change it to be so.

I don't think that this method actually gets called in generation, so changing it shouldn't matter much.

Reply:
Thank you for catching this; was meant as an option for future ROM reconfiguration option, for which I have yet to brainstorm a more likely use case than ``do not have access to the sharelink/settings object, and people cannot cooperate and choose a starting point that isn't Active/Wait``.